unraid_mcp/tools/unraid.py

```
@@ -0,0 +1,1891 @@
+                vms = [vms]
+            for vm in vms:
+                if vm.get("uuid") == vm_id or vm.get("id") == vm_id or vm.get("name") == vm_id:
+                    return dict(vm)
+            available = [f"{v.get('name')} (UUID: {v.get('uuid')})" for v in vms]
+            raise ToolError(f"VM '{vm_id}' not found. Available: {', '.join(available)}")
+
+        data = await make_graphql_request(_VM_MUTATIONS[subaction], {"id": vm_id})
+        field = _VM_MUTATION_FIELDS.get(subaction, subaction)
+        if data.get("vm") and field in data["vm"]:
+            return {"success": data["vm"][field], "subaction": subaction, "vm_id": vm_id}
+        raise ToolError(f"Failed to {subaction} VM or unexpected response")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)